lib/rubocop/cop/style/redundant_each.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for redundant `each`.
+      #
+      # @safety
+      #   This cop is unsafe, as it can produce false positives if the receiver
+      #   is not an `Enumerable`.
+      #
+      # @example
+      #
+      #   # bad
+      #   array.each.each { |v| do_something(v) }
+      #
+      #   # good
+      #   array.each { |v| do_something(v) }
+      #
+      #   # bad
+      #   array.each.each_with_index { |v, i| do_something(v, i) }
+      #
+      #   # good
+      #   array.each.with_index { |v, i| do_something(v, i) }
```

Comment:
I'm guessing that using `each_with_index` directly is also good. Same with `each_with_object`. Probably we'll need some separate cop about such iterators (e.g. something suggesting `each.with_object` over `each_ with_object`).

Reply:
Sure,`each_with_index` is also accepted. I've updated it!
OTOH, I did a quick check and saw no difference in performance. This autocorrection chooses `each.with_object` because the receiver is an `Enumerator` and using method chain. But, I'm not sure which of `each_with_object` and `each.with_object` is definitely better as the default when receiver is a collection object (e.g. instance of Array, Hash, Set).